CMakeLists.txt

```diff
@@ -1,124 +1,232 @@
     src/parse_yml.cpp
     src/camera_parameters.cpp
 )
-target_link_libraries(camera_calibration_parser ${catkin_LIBRARIES})
+target_include_directories(camera_calibration_parser PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>
+)
+target_link_libraries(camera_calibration_parser yaml_utils::yaml_utils)
+ament_target_dependencies(camera_calibration_parser
+    sensor_msgs
+)
+
 
 add_executable(load_camera_config_test src/load_camera_config_test.cpp)
+target_include_directories(load_camera_config_test PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>
+)
 target_link_libraries(load_camera_config_test camera_calibration_parser)
+ament_target_dependencies(load_camera_config_test sensor_msgs)
 
 
 # Python Bindings
-catkin_add_pybind11_module(py_camera_types opencv_driver ${pylon_driver})
-catkin_add_pybind11_module(py_tricamera_types
-    ${OpenCV_LIBRARIES} ${tricamera_driver} pybullet_tricamera_driver)
+add_pybind11_module(py_camera_types srcpy/py_camera_types.cpp
+    LINK_LIBRARIES
+        camera_observations
+        opencv_driver
+        ${pylon_driver}
+)
+add_pybind11_module(py_tricamera_types srcpy/py_tricamera_types.cpp
+    LINK_LIBRARIES
+        ${OpenCV_LIBRARIES}
+        ${tricamera_driver}
+        pybullet_tricamera_driver
+)
 
 
 # Installation
 install(DIRECTORY include/${PROJECT_NAME}/
-        DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION})
+        DESTINATION include/${PROJECT_NAME})
 
 install(DIRECTORY config/
-        DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}/config)
+        DESTINATION share/${PROJECT_NAME}/config)
+
+install(FILES cmake/FindPylon.cmake
+        DESTINATION share/${PROJECT_NAME}/cmake)
 
 install(
-    TARGETS opencv_driver ${pylon_driver} ${tricamera_driver}
-    ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-    LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-    RUNTIME DESTINATION ${CATKIN_GLOBAL_BIN_DESTINATION}
+    TARGETS
+        ${pylon_driver}
+        ${tricamera_driver}
+        camera_calibration_parser
+        camera_observations
+        opencv_driver
+        pybullet_tricamera_driver
+    EXPORT export_${PROJECT_NAME}
+    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION lib
+    RUNTIME DESTINATION bin
+)
+
+install(
+    PROGRAMS
+        scripts/analyze_tricamera_log.py
+        scripts/calibrate_cameras.py
+        scripts/calibrate_trifingerpro_cameras.py
+        scripts/camera_log_viewer.py
+        scripts/charuco_board.py
+        scripts/convert_to_camera_pose.py
+        scripts/detect_aruco_marker.py
+        scripts/overlay_camera_stream.py
+        scripts/record_image_dataset.py
+        scripts/rodrigues_to_quaternion.py
+        scripts/tricamera_log_converter.py
+        scripts/tricamera_log_viewer.py
+        scripts/tricamera_test_connection.py
+        scripts/verify_calibration.py
+    DESTINATION lib/${PROJECT_NAME}
```

Comment:
I would install the files, renamimg them in order to remove the .py and set the executable setting during install